openmetadata-ui/src/main/resources/ui/src/components/common/UserSelectableList/UserSelectableList.interface.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { PopoverProps } from 'antd';
+import { EntityReference } from 'generated/entity/teams/user';
+import { ReactNode } from 'react';
+
+export type UserSelectableListProps =
+  | {
+      hasPermission: boolean;
+      selectedUsers: EntityReference[];
+      children?: ReactNode;
+      popoverProps?: PopoverProps;
+    } & (
+      | {
+          multiSelect?: true;
+          onUpdate: (updatedUsers: EntityReference[]) => void;
+        }
+      | {
+          multiSelect: false;
+          onUpdate: (updatedUsers: EntityReference) => void;
+        }
+    );
```

Comment:
```suggestion
export type UserSelectableListProps {
  hasPermission: boolean;
  selectedUsers: EntityReference[];
  onUpdate: (updatedUsers: EntityReference | EntityReference[]) => void;
  multiSelect?: boolean;
  children?: ReactNode;
  popoverProps?: PopoverProps;
}
```

Reply:
It won't work for conditional props restriction on parent level